### ISSUE TITLE
After removing not used images (docker rmi) swarmpit started to show this error

### ISSUE THREAD

**Issue by edpichler** [swarmpit/swarmpit#508]
I removed docker images because the server was getting out of space with the comand `docker rmi `(without forcing). After that, when I try to edit a service in swarmpit and press deploy, this error is shown:

![image](https://user-images.githubusercontent.com/358696/82470490-bab6d300-9ac5-11ea-8964-0216937178c5.png)

I pulled images to see if it links with the server again but no success. How could I fix this error?

I use swarmpit v1.9.

**Comment by nohaapav** [comment]
@edpichler Are you sure you have your registry linked with swarmpit ? Go to registries and you should see record matching given `analogic` registry

**Comment by edpichler** [comment]
@nohaapav I have the public Docker hub registry. Same as always.

The interesting is that when I deactivate the stack and edit the compose file using the UI, the deployment works! But if I edit the service, it doesn't. Even right after deploying the compose file.

**Comment by nohaapav** [comment]
Is it linked with swarmpit ? Your dockerhub account ?

The thing is:
For stack deployment Swarmpit is using `docker cli`
For rest Swarmpit is using `docker api`

That means if you're logged in on your VM stack api works as you have credentials stored in cred-helper, but service don't as we can't generate token in order to fetch registry data.

**Comment by edpichler** [comment]
> Is it linked with swarmpit ? Your dockerhub account ?
Yes. Everything I am doing is in swarm bit. The only thing I did outside of swarmpit was deleting not used images (`docker rmi`)

**Comment by nohaapav** [comment]
Can you provide full name of image ? it's strange that error is saying `analogic` registry was not found.

**Comment by edpichler** [comment]
Sure. `analogic/poste.io:latest`

Super strange. This is happening with all images.
To recap: 
- I had the version 1.8, and I installed many stacks.
- Used it for months, and many edits on services using the UI.
- Upgraded swarmpit to 1.9.
- Got out of space. Removed images not being used with docker rmi (not forcing).
- Now, I edit anything, like a port of a service and try to deploy, this error is shown.

**Comment by edpichler** [comment]
It seems the services lost the link with the registry. I will try to create a new stack to test editing it.

**Comment by edpichler** [comment]
I tested it. I created a new stack, deployed, then when trying to edit it, same error.

**Comment by nohaapav** [comment]
@edpichler Do you know what the problem is ? dot `.` in image name .. This is the following logic how dockerhub image is distinguished: 
https://github.com/swarmpit/swarmpit/blob/master/src/cljc/swarmpit/docker/utils.cljc#L38

Swarmpit thinks that analogic is name of registry not dockerhub image and trying to find such registry in swarmpit DB.

We have to fix this logic, in meantime you can rename the image in order to make it work. For instance `analogic/mail-server` is working fine.

Strange that this logic is in codebase from the very beginning and nobody complained. Probably not many images matching names with dot.

**Comment by edpichler** [comment]
Tesded it now with nginx image. It worked! But with the analogic image, it does not! So the registry is liked correctly, the problem is when I delete an image.

**Comment by edpichler** [comment]
@nohaapav oh man, you found it! It makes sense. Strange that no one noticed it. Thank you.

**Comment by junebug12851** [comment]
I think I might be having a similar issue, all my images have dots in the names. In fact I have a public image with a dot in the name I just tested and it does the same thing.

I can edit the compose file for the whole stack and redeploy the whole stack, but I can't make any changes to a service individually or make any changes anywhere that uses the deployment wizard because the wizard rejects it and gives that error.

**Comment by nohaapav** [comment]
@junebug12851 true, you can't change service image via wizard at the moment so the only way to do so is via stack api.
